app_update_checker/lib/src/finder/update_finder.dart

```diff
@@ -0,0 +1,109 @@
+// ignore_for_file: avoid-unnecessary-reassignment, avoid-nested-switches, prefer-correct-identifier-length
+
+import '../localizer/models/release.dart';
+import '../localizer/models/update_config.dart';
+import '../shared/release_status.dart';
+import '../shared/update_platform.dart';
+import '../shared/version.dart';
+
+class UpdateFinder {
+  final Version appVersion;
+  final UpdatePlatform platform;
+
+  const UpdateFinder({
+    required this.appVersion,
+    required this.platform,
+  });
+
+  Release? findAvailableRelease(UpdateConfig config) {
+    final releases = config.releases;
+
+    // Sorted in descending order
+    releases.sort((a, b) => -a.version.compareTo(b.version));
+
+    Release? currentRelease;
+    Release? latestRelease;
+    ReleaseStatus latestReleaseStatus = ReleaseStatus.active;
+
+    releasesLoop:
+    for (final release in releases) {
+      if (release.version < appVersion) continue;

```

Comment:
Раз мы идём по убыванию, тогда здесь break можно поставить
```suggestion
      if (release.version < appVersion) break;

```